python/pythonlab/neighborhood/neighborhood/support/world.py

```diff
@@ -0,0 +1,27 @@
+from .grid_factory import GridFactory
+
+class World(object):
+  """
+  A singleton class that represents the world of the neighborhood.
+  The first time it is created it will set the grid to None. Users of the world
+  should check if the grid is None, and set it appropriately if it is.
+  All subsequent references will reuse the same grid, so multiple painters can operate
```

Comment:
Curious how multiple painters operate on same grid in javalab version as `World.java` is not a singleton class.

Reply:
discussed offline--short version is Javabuilder tracks the `World` in a `Context` object so there is a single `World` object shared between painters